servers/services/src/main/java/org/projectnessie/services/rest/BaseResource.java

```diff
@@ -87,17 +78,11 @@ protected BaseResource(
       }
 
       // we need to make sure that the hash in fact exists on the named ref
-      Hash hash = Hash.of(hashOnRef);
-      try (Stream<WithHash<CommitMeta>> commits = store.getCommits(namedRefWithHash.getValue())) {
-        if (commits.noneMatch(c -> c.getHash().equals(hash))) {
-          throw new NessieNotFoundException(
-              String.format("Could not find commit '%s' in reference '%s'.", hashOnRef, namedRef));
-        }
-      }
-      return WithHash.of(hash, namedRefWithHash.getValue());
+      return WithHash.of(
+          getStore().hashOnReference(namedRefWithHash.getValue(), Optional.of(Hash.of(hashOnRef))),
+          namedRefWithHash.getValue());
     } catch (ReferenceNotFoundException e) {
-      throw new NessieNotFoundException(
-          String.format("Could not find commit '%s' in reference '%s'.", hashOnRef, namedRef));
+      throw new NessieNotFoundException(e.getMessage());
```

Comment:
Unrelated to this PR, but WDYT about having a helper class to replace all these `new NesseXyzException(e.getMessage())` calls?

Reply:
yes that would definitely be useful. I saw that we already have similar things in `DatabaseAdapterUtil` but it would be better to have this at a more central place where `DatabaseAdapterUtil` and other components can use this.